tests/test_asset_fingerprinter.py

```diff
@@ -0,0 +1,86 @@
+from notifications_utils.asset_fingerprinter import AssetFingerprinter
+
+
+class TestAssetFingerprint:
+    def test_url_format(self, mocker):
+        get_file_content_mock = mocker.patch.object(AssetFingerprinter, "get_asset_file_contents")
+        get_file_content_mock.return_value = b"""
+            body {
+                font-family: nta;
+            }
+        """
+        asset_fingerprinter = AssetFingerprinter(asset_root="/suppliers/static/")
+        assert (
+            asset_fingerprinter.get_url("application.css")
+            == "/suppliers/static/application.css?418e6f4a6cdf1142e45c072ed3e1c90a"  # noqa
+        )
+        assert (
+            asset_fingerprinter.get_url("application-ie6.css")
+            == "/suppliers/static/application-ie6.css?418e6f4a6cdf1142e45c072ed3e1c90a"  # noqa
+        )
+
+    def test_building_file_path(self, mocker):
+        get_file_content_mock = mocker.patch.object(AssetFingerprinter, "get_asset_file_contents")
+        get_file_content_mock.return_value = b"""
+            document.write('Hello world!');
+        """
+        fingerprinter = AssetFingerprinter()
+        fingerprinter.get_url("javascripts/application.js")
+        fingerprinter.get_asset_file_contents.assert_called_with("app/static/javascripts/application.js")
+
+    def test_hashes_are_consistent(self, mocker):
+        get_file_content_mock = mocker.patch.object(AssetFingerprinter, "get_asset_file_contents")
+        get_file_content_mock.return_value = b"""
+            body {
+                font-family: nta;
+            }
+        """
+        asset_fingerprinter = AssetFingerprinter()
+        assert asset_fingerprinter.get_asset_fingerprint(
+            "application.css"
+        ) == asset_fingerprinter.get_asset_fingerprint("same_contents.css")
+
+    def test_hashes_are_different_for_different_files(self, mocker):
+        get_file_content_mock = mocker.patch.object(AssetFingerprinter, "get_asset_file_contents")
+        asset_fingerprinter = AssetFingerprinter()
+        get_file_content_mock.return_value = b"""
+            body {
+                font-family: nta;
+            }
+        """
+        css_hash = asset_fingerprinter.get_asset_fingerprint("application.css")
+        get_file_content_mock.return_value = b"""
+            document.write('Hello world!');
+        """
+        js_hash = asset_fingerprinter.get_asset_fingerprint("application.js")
+        assert js_hash != css_hash
+
+    def test_hash_gets_cached(self, mocker):
+        get_file_content_mock = mocker.patch.object(AssetFingerprinter, "get_asset_file_contents")
+        get_file_content_mock.return_value = b"""
+            body {
+                font-family: nta;
+            }
+        """
+        fingerprinter = AssetFingerprinter()
+        assert fingerprinter.get_url("application.css") == "/static/application.css?418e6f4a6cdf1142e45c072ed3e1c90a"
+        fingerprinter._cache["application.css"] = "a1a1a1"
```

Comment:
aren't we manually setting that cache here?

If we are, then what are we testing?

Reply:
I think it’s testing that the value returned is whatever’s in the cache, i.e. it isn’t recomputing it every time

Reply:
Ahh fair, so maybe the test name needs changing, for example to `test_get_fingerprint_from_cache` - because it's about getting a thing from cache, not it getting cached.

Reply:
Changed in [c205017](https://github.com/alphagov/notifications-utils/pull/1148/commits/c205017152448763612fc90dbf9a523c204279b4)